orderer/common/channelparticipation/validator.go

```diff
@@ -31,7 +31,11 @@ func ValidateJoinBlock(configBlock *cb.Block) (channelID string, err error) {
 		return "", errors.New("invalid block: does not have metadata")
 	}
 
-	if !bytes.Equal(protoutil.BlockDataHash(configBlock.Data), configBlock.Header.DataHash) {
+	dataHash, err := protoutil.BlockDataHash(configBlock.Data)
+	if err != nil {
+		return "", err
+	}
```

Comment:
Add a unit test for when that happens, please

Reply:
I removed it, because we only get config blocks so we cannot fail this check.